internal/service/ssm/parameter.go

```diff
@@ -112,6 +112,7 @@ func ResourceParameter() *schema.Resource {
 				Sensitive:    true,
 				Computed:     true,
 				ExactlyOneOf: []string{"insecure_value", "value"},
+				ValidateFunc: validation.StringLenBetween(0, 4096),				
```

Comment:
I think you missing the diffrentiation between tiers:
Standard max is 4k
Advanced/Intelligent-Tiering is 8k

you can see it here: https://docs.aws.amazon.com/systems-manager/latest/userguide/parameter-store-advanced-parameters.html